### PR TITLE
[oxygen] Merge forward from 2017.7 to oxygen

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - salt/modules/pip.py
  - salt/modules/win_pkg.py
  - salt/runners/manage.py
  - salt/states/pkg.py
  - salt/transport/zeromq.py
  - salt/utils/event.py
  - tests/integration/modules/test_pip.py
  - tests/integration/states/test_user.py
  - tests/unit/grains/test_core.py
  - tests/unit/states/test_archive.py
  - tests/unit/utils/test_boto.py

### CODE REVIEW

**Comment by rallytime** [comment]
re-run centos

**Comment by rallytime** [comment]
re-run centos

File: salt/modules/win_pkg.py [hunk 1]
```diff
@@ -1138,15 +1166,18 @@ def install(name=None, refresh=False, pkgs=None, **kwargs):
             version_num = six.text_type(version_num)
 
         if not version_num:
+            if pkg_name in old:
+                log.debug('A version ({0}) already installed for package '
+                          '{1}'.format(version_num, pkg_name))
```
Comment (terminalmage) [inline salt/modules/win_pkg.py]: This log message needs to be fixed in Oxygen, `.format()` changed to string replacement.

**Review by terminalmage — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
I'll need to redo this because it has a merge conflict now.

**Pull request closed without merging**